proposed/0000-Use-public-key-for-id-on-all-messages.md

```diff
@@ -0,0 +1,60 @@
+- Feature Name: Use public key alone to authorise actions 
+- Type Enhancement
+- Related components routing, maidsafe_types, maidsafe_client, maidsafe_vaults
+- Start Date: 23-06-2015
+- RFC PR: 
+- Issue number: 
+
+# Summary
+
+At the moment client Id packets are create and a hash of these packets used as identity. 
+This involves storing such packets prior to the client interacting with the network. 
+As such this requires an initial `unauthorised put` as the client is not known to the 
+network and cannot be recognised without this. It also means there are many lookups
+to link the hash of the id packet with the client taking the action. 
+These lookups are costly it is proposed here that they are not required.  
+
+# Motivation
+
+Storing Id packets is a strain on the network. Lookups are a strain and will slow down
+network actions. Id packets themselves may be a security issue (can they be hacked), although
+highly unlikely, it is impossible to hack if they do not exist! 
+
+This mechanism does not reduce the total number of clients or types of clients as they all exist 
+in an address space of 2^256 and app developers can use multiple Id's in very clever ways, such 
+as the SAFE network does not distinguish between public id's and private id's. Now these and more are possible.
+Id's could be created to share data types (as co-owners) and allow shares access and write capability
+between applications and groups. 
+
+# Detailed design
+
+As clients `Put` data or messages on the network (i.e. ask to create) they send such requests via the 
+ClientManager type persona (MaidManager in the case of data). These requests are signed and include the
+clients `PublicKey`. This public key can be allocated, or matched to an existing,to a client account. 
+The ClientManager may then take any action, such as authorising the `Put`, taking a balance etc. and all against the key. 
+
+If a client wishes to amend any data (structured data) then this is already signed by that client and contains the 
+`PublicKey` The client will sign the request to alter the data (by overwriting with a new valid one) and again
+the public key of the signed message is included in the message, confirming what key made the request. This key
+can then be confirmed to be one of the owners of the `StructuredData` element and if multiple owners
+then the signatures of the data element being uploaded is checked against the list of signatures of the new chunk
+using the owners keys of the last chunk. 
```

Comment:
This also puts responsability on the client/app for multisig to negotiate amongst the owners to agree on a new version and delegate one client to collect all signatures and execute the put on behalf of all;

alternatively we can also put such behaviour in "Sentinel-like accumulator", where clients would have to coordinate their put; this allows a double route though (as it would not exclude the above), and maybe thats not a desirable effect; also Sentinel should not be involved here as the core idea is for the put request to be self-verifiying